craft_parts/executor/step_handler.py

```diff
@@ -288,32 +293,63 @@ def _handle_control_api(self, scriptlet_name: str, function_call: str) -> None:
                     f"{scriptlet_name!r} control call missing attribute {attr!r}"
                 )
 
-        function_name = function_json["function"]
-        function_args = function_json["args"]
+        cmd_name = function_json["function"]
+        cmd_args = function_json["args"]
+
+        return self._process_api_commands(
+            cmd_name, cmd_args, step=step, scriptlet_name=scriptlet_name
+        )
+
+    def _process_api_commands(
```

Comment:
this method does not seem to gain us much as _handle_control_api is mostly a passthrough for this

Reply:
Split into a new method to reduce complexity (linters also complained about that -- could disable the warning and coalesce back into a single method, but having it separated looks better for organization)